substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/management/SubstrateThreadMXBean.java

```diff
@@ -150,45 +152,61 @@ public int getDaemonThreadCount() {
         return daemonThreadCount.get();
     }
 
-    /* All remaining methods are unsupported on Substrate VM. */
-
     @Override
     public long[] getAllThreadIds() {
-        return new long[0];
+        return Arrays.stream(PlatformThreads.getAllThreads())
+                        .mapToLong(Thread::threadId)
+                        .toArray();
+    }
+
+    public static Thread getThreadById(long id) {
+        return Arrays.stream(PlatformThreads.getAllThreads())
+                        .filter(thread -> thread.threadId() == id)
+                        .findAny().orElse(null);
     }
 
     @Override
     public ThreadInfo getThreadInfo(long id) {
-        return null;
+        return getThreadInfo(id, -1);
     }
 
     @Override
     public ThreadInfo[] getThreadInfo(long[] ids) {
-        return new ThreadInfo[0];
+        return (ThreadInfo[]) Arrays.stream(ids).mapToObj(this::getThreadInfo).toArray();
```

Comment:
Can you re-use `getThreadInfo(long[] ids, int maxDepth)` here?
```suggestion
        return getThreadInfo(ids, -1);
```